src/logic/createFormControl.ts

```diff
@@ -1192,9 +1192,11 @@ export function createFormControl<
     });
   };
 
-  const setFocus: UseFormSetFocus<TFieldValues> = (name) => {
+  const setFocus: UseFormSetFocus<TFieldValues> = (name, options = {}) => {
     const field = get(_fields, name)._f;
-    (field.ref.focus ? field.ref : field.refs[0]).focus();
+    options.shouldSelect
+      ? (field.ref.select ? field.ref : field.refs[0]).select()
+      : (field.ref.focus ? field.ref : field.refs[0]).focus();
```

Comment:
```suggestion
    const field = field.ref.select ? field.ref : field.refs[0];
    options.shouldSelect ? field.select() : field.focus();
```
It may use fewer bytes.

Reply:
You mean, like that? We still need to get the `field`
```js
    const field = get(_fields, name)._f.ref.select
      ? get(_fields, name)._f.ref
      : get(_fields, name)._f.refs[0];
    options.shouldSelect ? field.select() : field.focus();
```

Reply:
yea, just want to test it out, if it uses fewer bytes for the bundle size.

Reply:
oh nvm, i think my suggestion above is invalid.

Reply:
let me try another one.

Reply:
```
const setFocus: UseFormSetFocus<TFieldValues> = (name, options) => {
    const field = get(_fields, name)._f;
    const fieldRef = field.refs ? field.refs[0] : field.ref;
    options.shouldSelect
      ? fieldRef.select()
      : fieldRef.focus();
  };
```

<img width="509" alt="Screen Shot 2022-02-10 at 10 15 39 am" src="https://user-images.githubusercontent.com/10513364/153306591-a18fbbb8-a5f4-412f-83ba-d0801771d16b.png">

goes to `8.51`

Reply:
<img width="365" alt="image" src="https://user-images.githubusercontent.com/16290753/153307097-9863925c-1a3c-4bf9-a594-b83540d4a9c6.png">

Reply:
Updated